components/brave_wallet/browser/internal/hd_key_zip32.cc

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_wallet/browser/internal/hd_key_zip32.h"
+
+#include <utility>
+
+#include "base/check.h"
+
+static_assert(BUILDFLAG(ENABLE_ORCHARD));
+
+namespace brave_wallet {
+
+HDKeyZip32::HDKeyZip32(rust::Box<OrchardExtendedSpendingKeyResult> esk)
+    : extended_spending_key_(std::move(esk)) {}
+
+HDKeyZip32::~HDKeyZip32() = default;
+
+// static
+std::unique_ptr<HDKeyZip32> HDKeyZip32::GenerateFromSeed(
```

Comment:
what's the reason for returning a unique_ptr here? Why does this necessarily have to be heap allocated? It would actually be better for security if both `HDKeyZip32` and the underlying `OrchardExtendedSpendingKey` were not heap allocated because memory on the heap is available to all threads in the process, but maybe that's unavoidable with rust ffi. cc @kdenhartog

Reply:
even without the small security benefit I'm not sure why this needs to be a unique_ptr. Is there a reason?

Reply:
it's not a huge deal, but not everything needs to be heap allocated and there is some small security benefit here

Reply:
nvm, this will eventually end up as part of something heap allocated anyway